src/worker/ingestion/action-manager.ts

```diff
@@ -0,0 +1,63 @@
+import { Action, PluginsServerConfig } from '../../types'
+import { DB } from '../../utils/db/db'
+import { status } from '../../utils/status'
+
+type ActionCache = Record<Action['id'], Action>
+
+export class ActionManager {
+    private ready: boolean
+    private db: DB
+    private actionCache: ActionCache
+
+    constructor(db: DB) {
+        this.ready = false
+        this.db = db
+        this.actionCache = {}
+    }
+
+    public async prepare(): Promise<void> {
+        this.actionCache = await this.db.fetchAllActionsMap()
```

Comment:
Not sure how exactly action manager will be used yet, so:

If something goes wrong with fetching actions, such that this cache isn't populated, do we want to stop ingestion completely? or continue?

Reply:
Well, in that case an error will be thrown (if e.g. Postgres connection fails) and indeed ingestion will be stopped

Reply:
Right, I'm asking if that's what we ought to do, or not?

Reply:
Ah, yes, we do, since otherwise data integrity will be at risk (action matching wouldn't work). Besides, this is not likely to fail for internal reasons, and if Postgres (the external dependency) fails, then we are in catastrophic failure territory anyway